### PR TITLE
Scheduling waste metrics

### PR DESCRIPTION
**Pull request by onursatici**

### CODE REVIEW

File: internal/metrics/waste.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	demandFulfilledTime time.Time
+	demandCreationTime  time.Time
+}
+
+type podKey struct {
+	Namespace string
+	Name      string
+}
+
+type schedulingInfo map[podKey]demandInfo
```
Comment (laflechejonathan) [inline internal/metrics/waste.go:113]: rename to `demandsByPod`

File: internal/metrics/waste.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	}
+	totalTimeNoDemand = tagInfo{
+		tag:          metrics.MustNewTag(schedulingWasteTypeTagName, "total-time-no-demand"),
+		logThreshold: 10 * time.Minute,
+	}
+)
+
+type wasteMetricsReporter struct {
+	ctx  context.Context
+	info schedulingInfo
```
Comment (laflechejonathan) [inline internal/metrics/waste.go:53]: rename to demandsByPod

File: internal/metrics/waste.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+			)
+			delete(r.info, key)
+		}
+	}
+
+}
+
+type tagInfo struct {
+	tag          metrics.Tag
+	logThreshold time.Duration
```
Comment (laflechejonathan) [inline internal/metrics/waste.go:164]: rename to `slowLogThreshold`

File: internal/metrics/waste.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+func (r *wasteMetricsReporter) onDemandFulfilled(demand *v1alpha1.Demand) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	r.info[podKey{demand.Namespace, utils.PodName(demand)}] = demandInfo{
+		demandFulfilledTime: time.Now(),
+		demandCreationTime:  demand.CreationTimestamp.Time,
+	}
+}
+
+func (r *wasteMetricsReporter) cleanup() {
```
Comment (laflechejonathan) [inline internal/metrics/waste.go:146]: cleanupDemandCache

File: internal/metrics/waste.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+	podInformer coreinformers.PodInformer,
+	demandInformer *crd.LazyDemandInformer,
+) {
+	reporter := &wasteMetricsReporter{
+		ctx:  ctx,
+		info: make(schedulingInfo),
+	}
+
+	podInformer.Informer().AddEventHandler(
+		clientcache.FilteringResourceEventHandler{
```
[the remark this answers is not in the document]
Comment (onursatici) [reply internal/metrics/waste.go:70]: it is a k8s client-go thing

**Pull request merged**